### PR TITLE
Use AsyncDnsCache directly in DnsServiceDiscovery, #25948

### PR DESCRIPTION
**Pull request by patriknw**
* to avoid (2) asks

Slightly different approach to access the cache than the options suggested in the ticket.

Refs #25948

### CODE REVIEW

File: akka-discovery/src/main/scala/akka/discovery/dns/DnsServiceDiscovery.scala [hunk 1]
```diff
@@ -72,39 +80,99 @@ private[akka] class DnsServiceDiscovery(system: ExtendedActorSystem) extends Ser
...
 
   override def lookup(lookup: Lookup, resolveTimeout: FiniteDuration): Future[Resolved] = {
-    if (lookup.portName.isDefined && lookup.protocol.isDefined) {
-      val srvRequest = s"_${lookup.portName.get}._${lookup.protocol.get}.${lookup.serviceName}"
-      log.debug("Lookup [{}] translated to SRV query [{}] as contains portName and protocol", lookup, srvRequest)
-      dns.ask(DnsProtocol.Resolve(srvRequest, Srv))(resolveTimeout).map {
+    if (lookup.portName.isDefined && lookup.protocol.isDefined)
+      lookupSrv(lookup, resolveTimeout)
+    else
+      lookupIp(lookup, resolveTimeout)
```
Comment (chbatey) [inline akka-discovery/src/main/scala/akka/discovery/dns/DnsServiceDiscovery.scala]: lookupArecord?

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-discovery/src/main/scala/akka/discovery/dns/DnsServiceDiscovery.scala (hunk 1 above)
Comment (chbatey) [reply akka-discovery/src/main/scala/akka/discovery/dns/DnsServiceDiscovery.scala]: Ah i see we call it Ip

**Review by raboof — APPROVED** [review]
Looks good, should indeed save on the `ask` invocations.

I suspect this could be optimized further by detecting when there is already a resolve 'in-flight' for a request, but that can be done later.

**Pull request merged**